viime/views.py

```diff
@@ -724,6 +725,56 @@ def get_pca_overview(validated_table: ValidatedMetaboliteTable):
     return Response(png_content, mimetype='image/png')
 
 
+@csv_bp.route('/csv/<uuid:csv_id>/analyses/plsda', methods=['GET'])
+@use_kwargs({
+    'num_of_components': fields.Integer(required=False)
+})
+@load_validated_csv_file
+def get_plsda(validated_table: ValidatedMetaboliteTable, num_of_components: Optional[int] = None):
+    measurements = validated_table.measurements
+    groups = validated_table.groups
+
+    rows, cols = measurements.shape[:2]
+    max_components = min(rows, cols)
+    if num_of_components is None:
+        num_of_components = max_components
+    if num_of_components > max_components:
+        return jsonify({
+            'error': 'invalid num_of_components'
+        }), 400
+
+    scores = plsda(measurements, groups, num_of_components, 'scores')
+    loadings = plsda(measurements, groups, num_of_components, 'loadings')
```

Comment:
I looked into this, but the scores and loadings don't fit into the same R `dataframe` since they are different sizes. The PCA analysis also uses two opencpu requests to retrieve its scores and loadings, but in that case there is two separate flask endpoints -
`GET /csv/<uuid:csv_id>/plot/pca`
`GET /csv/<uuid:csv_id>/plot/loadings`

I can split the PLS-DA scores and loadings into two endpoints like that, but I believe they will always be called at the same time anyway so performance will be the same. What do you think?

Reply:
I see.  I think there are ways of returning more unstructured data types, but I never instrumented it.  This is fine for now.